docs/DesignDocs/oe_log_message()_callback_proposal.md

```diff
@@ -0,0 +1,61 @@
+Proposal of Public Callback API for oe_log_message()
+====
+
+# Motivation
+
+This document describes the proposed public callback API for oe_log_message().
+
+There are several problems about the current logging that prevent end users to
+make full use of log:
+- It does't have mechanism to communicate between enclave and host.
+- Message can't be redirected to different destinations on demand.
+- Lack of method for end user to customize messages.
+
+The callback API is supposed to solve such problems.
+
+# User Experience
+
+By adding this public API, user can customize the output of the log message:
+- The output destination can be stdout/stderr or any specified files, and it is
+adjustable during runtime.
+- The message can be adjusted according to the log level.
+- The user can get as much information about the message as internel API. For
```

Comment:
Does it just mean that there is no information gap between internal and public logging anymore?

Reply:
Yes all the information are exposed to end users.